### PR TITLE
feat(openapi-tooling): Add support for `oneOf` in generated clients

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

While testing the type generation with the spec I created in #22205 , I was running into weird output like,
```
import { PermissionCriteria } from '../models/PermissionCriteria.model';

export interface Authorize200ResponseItemsInner {
  result: Authorize200ResponseItemsInnerResultEnum;
  pluginId: string;
  resourceType: string;
  conditions: PermissionCriteria;
  id: string;
}

export type Authorize200ResponseItemsInnerResultEnum = 'ALLOW' | 'DENY';
```
where I would expect a union type with references to each `oneOf` model like so, 
```
export type Authorize200ResponseItemsInner = ConditionalPolicyDecision | DefinitivePolicyDecision;
```

After some looking, the template that I cloned our stuff from only support basic schemas, but some other typescript templates support more complex schemas. The updates schemas are pulled from `typescript-angular` instead of the basic `typescript`. With this change, that becomes
```
export type Authorize200ResponseItemsInner =
  | ConditionalPolicyDecision
  | DefinitivePolicyDecision;
```
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{{#models}}
```
Comment (Rugvip) [inline packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache:1]: Could we add a `{{! ` comment stating where these files are from? As well as a [notice](https://github.com/backstage/backstage/blob/master/NOTICE) that highlights the upstream licences that we're subject to?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 👍 

Just wanna make sure get the licensing right here, since this is all copyrighted and licensed work.

File: packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache (hunk 1 above)
Comment (sennyeya) [reply packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache:1]: Should I update that top level notice or create a new template notice?

Comment (Rugvip) [reply packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache:3]: yep the top-level one

Comment (sennyeya) [reply packages/repo-tools/templates/typescript-backstage/modelGeneric.mustache:3]: Added in df7450e78cb0561926881f6a0b06b3f66c1c06e9

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍

**Pull request merged**